### PR TITLE
Feat/async

### PR DESCRIPTION
**Pull request by anserahmed87**

### CODE REVIEW

File: src/components/newsitemlist/index.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import React from 'react';
 import {NewsItem} from "../";
+import App from "../../App";
 
 export const NewsItemList = ({ids}) => {
+    //console.log(ids)
```
Comment (iamstarkov) [inline src/components/newsitemlist/index.js]: maybe you don't need this anymore

File: src/components/page-news-list/index.js [hunk 2]
```diff
@@ -1,7 +1,43 @@
...
+        const { ids } = this.state;
+        if (!ids) {
+            return <div>Loading?</div>
+        }
+        return (
+            <div>
+                <button onClick={this.fetchItems}>
+                   Refresh
+                </button>
+                <NewsItemList ids={ids}/>;
```
Comment (iamstarkov) [inline src/components/page-news-list/index.js]: you sure about this semicolon?

File: src/utils/api.js [hunk 3]
```diff
@@ -7,112 +7,24 @@
...
+   return fetchJson(`${BASE_URL}/item/${id}.json`);
+}
+
 
 export const api = {
-    getItemsIds: () => itemsIds,
-    getItem: (id) => itemsPerId[id],
+    getItemsIds,
+    getItem,
+   // getItem: (id) => itemsPerId[id],
```
Comment (iamstarkov) [inline src/utils/api.js]: comments are dead code, strive to not having them. you have git to store prev versions of your code

**Review by iamstarkov — APPROVED** [review]
minor things, but LGTM otherwise

File: src/components/newsitemlist/index.js (hunk 1 above)
Comment (anserahmed87) [reply src/components/newsitemlist/index.js]: fixed

File: src/utils/api.js (hunk 3 above)
Comment (anserahmed87) [reply src/utils/api.js]: removed

File: src/components/page-news-list/index.js (hunk 2 above)
Comment (anserahmed87) [reply src/components/page-news-list/index.js]: removed

**Comment by iamstarkov** [comment]
merge it

**Pull request merged**